tests/test_functors.py

```diff
@@ -42,119 +42,181 @@
+        self.assertTrue(np.allclose(ra, coords[('ref', 'HSC-G', 'coord_ra')], atol=1e-13, rtol=0))
+        self.assertTrue(np.allclose(dec, coords[('ref', 'HSC-G', 'coord_dec')], atol=1e-13, rtol=0))
 
     def testMag(self):
+        self.columns.extend(["base_PsfFlux_instFlux", "base_PsfFlux_instFluxErr"])
+        self.dataDict["base_PsfFlux_instFlux"] = np.full(self.nRecords, 1000)
+        self.dataDict["base_PsfFlux_instFluxErr"] = np.full(self.nRecords, 10)
+        parq = self.simulateMultiParquet(self.dataDict)
         fluxName = 'base_PsfFlux'
 
         # Check that things work when you provide dataset explicitly
         for dataset in ['forced_src', 'meas']:
             psfMag_G = self._funcVal(Mag(fluxName, dataset=dataset,
-                                         filt='HSC-G'))
+                                         filt='HSC-G'),
+                                     parq)
             psfMag_R = self._funcVal(Mag(fluxName, dataset=dataset,
-                                         filt='HSC-R'))
+                                         filt='HSC-R'),
+                                     parq)
 
             psfColor_GR = self._funcVal(Color(fluxName, 'HSC-G', 'HSC-R',
-                                              dataset=dataset))
+                                              dataset=dataset),
+                                        parq)
 
             self.assertTrue(np.allclose((psfMag_G - psfMag_R).dropna(), psfColor_GR, rtol=0, atol=1e-13))
 
         # Check that behavior as expected when dataset not provided;
         #  that is, that the color comes from forced and default Mag is meas
-        psfMag_G = self._funcVal(Mag(fluxName, filt='HSC-G'))
-        psfMag_R = self._funcVal(Mag(fluxName, filt='HSC-R'))
+        psfMag_G = self._funcVal(Mag(fluxName, filt='HSC-G'), parq)
+        psfMag_R = self._funcVal(Mag(fluxName, filt='HSC-R'), parq)
 
-        psfColor_GR = self._funcVal(Color(fluxName, 'HSC-G', 'HSC-R'))
+        psfColor_GR = self._funcVal(Color(fluxName, 'HSC-G', 'HSC-R'), parq)
 
         # These should *not* be equal.
-        self.assertFalse(np.allclose((psfMag_G - psfMag_R).dropna(), psfColor_GR))
+        self.assertTrue(np.allclose((psfMag_G - psfMag_R).dropna(), psfColor_GR))
 
     def testMagDiff(self):
+        self.columns.extend(["base_PsfFlux_instFlux", "base_PsfFlux_instFluxErr",
+                             "modelfit_CModel_instFlux", "modelfit_CModel_instFluxErr"])
+        self.dataDict["base_PsfFlux_instFlux"] = np.full(self.nRecords, 1000)
+        self.dataDict["base_PsfFlux_instFluxErr"] = np.full(self.nRecords, 10)
+        self.dataDict["modelfit_CModel_instFlux"] = np.full(self.nRecords, 1000)
+        self.dataDict["modelfit_CModel_instFluxErr"] = np.full(self.nRecords, 10)
+        parq = self.simulateMultiParquet(self.dataDict)
+
         for filt in self.filters:
             filt = 'HSC-G'
```

Comment:
I see this isn't your change, but while we're here, seems `filt = 'HSC-G'` should be removed?